src/components/strategies/common/StrategyChartHistory.tsx

```diff
@@ -0,0 +1,187 @@
+      buy: {
+        min: min,
+        max: max,
+      },
+      sell: {
+        min: order1.min || '0',
+        max: order1.max || max,
+      },
+    };
+  } else {
+    return {
+      buy: {
+        min: order0.min || min,
+        max: order0.max || max,
+      },
+      sell: {
+        min: order1.min || min,
+        max: order1.max || max,
+      },
+    };
+  }
+};
+
+interface Props {
+  type: TradeTypes;
+  base: Token;
+  quote: Token;
+  order0: BaseOrder;
+  order1: BaseOrder;
+  isLimit?: { buy: boolean; sell: boolean };
+  direction?: 'buy' | 'sell'; // Only for disposable
+  spread?: string; // Only overlapping
+  readonly?: boolean;
+  marketPrice?: string;
+  onPriceUpdates?: OnPriceUpdates;
+}
+
+export const StrategyChartHistory: FC<Props> = (props) => {
+  const timeout = useRef<NodeJS.Timeout>();
+  const { base, quote, type, order0, order1 } = props;
+  const { priceStart, priceEnd } = useSearch({ strict: false }) as TradeSearch;
+  const { marketPrice: externalPrice } = useMarketPrice({ base, quote });
+
+  const marketPrice = props.marketPrice
+    ? Number(props.marketPrice)
+    : externalPrice;
+
+  const direction = (() => {
+    if (type === 'market') return 'none';
+    return props.direction;
+  })();
+
+  const [prices, setPrices] = useState({
+    buy: { min: order0.min || '0', max: order0.max || '0' },
+    sell: { min: order1.min || '0', max: order1.max || '0' },
+  });
+  const [bounds, setBounds] = useState(
+    getBounds(order0, order1, [], direction)
+  );
+
+  const updatePrices: OnPriceUpdates = ({ buy, sell }) => {
+    setPrices({ buy, sell });
+    if (timeout.current) clearTimeout(timeout.current);
+    timeout.current = setTimeout(() => {
+      props.onPriceUpdates?.({ buy, sell });
+    }, 200);
+  };
+
+  const { data, isPending, isError } = useGetTokenPriceHistory({
+    baseToken: base.address,
+    quoteToken: quote.address,
+    start: priceStart ?? defaultStart().toString(),
+    end: priceEnd ?? defaultEnd().toString(),
+  });
+
+  useEffect(() => {
+    setPrices({
+      buy: { min: order0.min || '0', max: order0.max || '0' },
+      sell: { min: order1.min || '0', max: order1.max || '0' },
+    });
+  }, [order0.min, order0.max, order1.min, order1.max]);
+
+  useEffect(() => {
+    setBounds(getBounds(order0, order1, data, direction));
+  }, [order0, order1, data, direction]);
+
+  const priceChartType = config.ui.priceChart;
+  if (priceChartType === 'tradingView') {
+    return <TradingviewChart base={base} quote={quote} />;
+  }
+  if (isPending) {
+    return (
+      <section className="rounded-12 grid flex-1 items-center bg-black">
+        <CarbonLogoLoading className="h-[80px]" />
+      </section>
+    );
+  }
+  if (isError) {
+    return (
+      <NotFound
+        variant="error"
+        title="Well, this doesn't happen often..."
+        text="Unfortunately, price history for this pair is not available."
+      />
+    );
+  }
+  if (!data?.length) return null;
```

Comment:
Could you check this condition with isError and show the NotFound component for it?